src/running_tracker.rs

```diff
@@ -10,34 +10,26 @@ lazy_static! {
 }
 
 pub struct RunningTracker {
-    running: Arc<AtomicBool>,
     exit_code: Arc<AtomicI32>,
 }
 
 impl RunningTracker {
     fn new() -> Self {
         Self {
-            running: Arc::new(AtomicBool::new(true)),
             exit_code: Arc::new(AtomicI32::new(0)),
         }
     }
 
     pub fn quit(&self, reason: &str) {
-        self.running.store(false, Ordering::Relaxed);
         info!("Quit {}", reason);
     }
 
     pub fn quit_with_code(&self, code: i32, reason: &str) {
-        self.exit_code.store(code, Ordering::Relaxed);
-        self.running.store(false, Ordering::Relaxed);
+        self.exit_code.store(code, Ordering::Release);
         info!("Quit with code {}: {}", code, reason);
     }
 
-    pub fn is_running(&self) -> bool {
-        self.running.load(Ordering::Relaxed)
-    }
```

Comment:
The running tracker only tracks the exit code now, so maybe it could be renamed or replaced by variable. But I left that for future refactoring.